### ISSUE TITLE
More examples on selecting nested data

### ISSUE THREAD

**Issue by SalaryUltraman** [jmespath/jmespath.site#7]
Hello, it'd be appreciated if you could provide guidance on this one. I'd like to obtain the "general" object for the ID == 100.Thanks a lot in advance!

```
{
  "people": [
    {
      "general": {
        "id": "100",
        "age": 20,
        "other": "foo",
        "name": "Bob"
      },
      "history": {
        "first_login": "2014-01-01",
        "last_login": "2014-01-02"
      }
    },
    {
      "general": {
        "id": "101",
        "age": 30,
        "other": "bar",
        "name": "Bill"
      },
      "history": {
        "first_login": "2014-05-01",
        "last_login": "2014-05-02"
      }
    }
  ]
}
```

**Comment by jamesls** [comment]
Yeah no problem I'll get some examples added.  Thanks for the request.

**Comment by jamesls** [comment]
The expression you can use is ``people[?general.id==`100`].general | [0]``

I've added this example to the JMESPath examples page that walks through this expression: http://jmespath.org/examples.html#filtering-and-selecting-nested-data

Let me know if you have any other questions.

**Comment by SalaryUltraman** [comment]
Thanks a lot for the answer and the great explanation on the Examples page!

One other thing that I realized was at fault in my case: the records ID are expressed as strings instead of integers. (like in my question above: "id": "100"). This is not something I have direct control on as the JSON is provided by an external entity. So I realized that I actually need to write
```people[?general.id==`"100"`].general | [0]```

Maybe I'd suggest to put this kind of distinction in the tutorial page.

Thanks again for the help!
